include/seastar/core/gate.hh

```diff
@@ -47,21 +51,36 @@ public:
 class gate {
     size_t _count = 0;
     std::optional<promise<>> _stopped;
+
+#ifdef GATE_HOLDER_DEBUG
+    size_t _holders = 0;
+
+    void assert_not_held(const gate& g) const noexcept {
```

Comment:
probably could just take advantage of the fact that this is a member function? like
```c++
void assert_not_held() const noexcept {
    assert(!_holders && "gate moved with outstanding holders");
}
```

Reply:
good point. will do